components/script/script_thread.rs

```diff
@@ -686,6 +693,24 @@ pub struct ScriptThread {
     webgpu_port: RefCell<Option<Receiver<WebGPUMsg>>>,
 }
 
+struct BHMExitSignal {
+    closing: Arc<AtomicBool>,
+    js_context: ContextForRequestInterrupt,
+}
+
+impl BackgroundHangMonitorExitSignal for BHMExitSignal {
+    fn signal_to_exit(&self) {
+        // Note: can we be sure the interrupt will see `true`?
```

Comment:
I don't really know what this question means. If the interrupt uses the appropriate atomic operation sequencing, then it should.